cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -2385,6 +2385,10 @@ int getPodIndexFromPvcName(String pvcName)  {
                     Annotations.annotations(eoDeployment.getSpec().getTemplate()).put(
                             Ca.ANNO_STRIMZI_IO_CLIENTS_CA_CERT_GENERATION, String.valueOf(clientsCaCertGeneration));
                     return withVoid(deploymentOperations.reconcile(namespace, EntityOperator.entityOperatorName(name), eoDeployment));
+                })
+                .compose(dep -> {
```

Comment:
The entity operator might be disabled ... in that case the deployment will not exist ... I think we might need to handle it somehow? E.g. something like `if (this.entityOperator != null && eoDeployment != null) {` (look at the `entityOperatorDeployment` method).